### PR TITLE
#3115 Workshop shows stale brick detail after save

### LINKED ISSUE #3115: Workshop shows stale brick detail  after save
Context
---
- Likely due to RTK not being invalidated on save? Need to add steps to reproduce. We might just need to not cache the brick edit page in the workshop? Or we might be missing an invalidation?

Steps to Reproduce
---
- Open a blueprint in the workshop
- Modify it
- Return to the main workshop screen
- Select the same blueprint

Expected Result: see the new version of the blueprint
Actual Result: see the old version of the blueprint

### PR DESCRIPTION
**Pull request by BALEHOK**
## What does this PR do?

- Fixes #3115 
- Use RTKQ to create/update/delete packages

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/errors/networkErrorHelpers.ts [hunk 1]
```diff
@@ -29,7 +29,10 @@ export type SerializableAxiosError = Except<AxiosError, "toJSON">;
 
 // Copy of axios.isAxiosError, without risking to import the whole untreeshakeable axios library
 export function isAxiosError(error: unknown): error is SerializableAxiosError {
-  if (isObject(error) && Boolean(error.isAxiosError)) {
+  if (
+    isObject(error) &&
+    (Boolean(error.isAxiosError) || error.name === "AxiosError")
```
Comment (fregante) [inline src/errors/networkErrorHelpers.ts:34]: How come this wasn't working? What does the object received in this case look like?

Comment (BALEHOK) [reply src/errors/networkErrorHelpers.ts:34]: @fregante  I've changed the way we call the package API. It now goes through RTK Query:
https://github.com/pixiebrix/pixiebrix-extension/pull/4112/files#diff-c762c8cdc72a46df67815a73d7fdd45ac2264e8312211ab5f2fbe55733549f7bR74
`await deletePackage({ id }).unwrap()`

The error object is not an instance of `AxiosError` and doesn't provide `isAxiosError` property. Maybe that is related to the `unwrap` call.

What the error used to be:
Screen Shot 2022-08-24 at 4 53 02 PM

What the error is now (notice the prototype is a plain object):
Screen Shot 2022-08-24 at 4 54 23 PM

Comment (BLoe) [reply src/errors/networkErrorHelpers.ts:34]: It looks like we possibly could mess with our RTK config/setup to get the right error type information in there?

https://github.com/rtk-incubator/rtk-query/issues/86#issuecomment-738845312

Comment (BALEHOK) [reply src/errors/networkErrorHelpers.ts:34]: I can dig into the error serialization with RTK Query and unwrapping.

Comment (BALEHOK) [reply src/errors/networkErrorHelpers.ts:34]: It is the `serializeError` that converts `AxiosError` into a plain object. The library description reads:
`Serialize/deserialize an error into a plain object`.

We have to serialize `AxiosError` to store it in redux. RTK mutation throws exactly the object returned by the `appBaseQuery`. That is what we get with `await deletePackage({ id }).unwrap()`.
To convert it to an instance of `AxiosError` we need to call `deserialize`, this usually happens with a call to [selectError](https://github.com/pixiebrix/pixiebrix-extension/blob/3c62042ebd5af6d5d415915e89724352e9072fb5/src/errors/errorHelpers.ts#L303).

For non-RTK Query error there's no such issue. A direct call to axios client throws non-serialized `AxiosError`.

The bottom line, RTK throws a plain object. More and more often we will be dealing with such errors. `error.isAxiosError` is not defined for them, hence our logic to get the error message will not work as expected.

I see the following options:
1. Keep the check for `error.name === "AxiosError"`. This covers the case of serialized errors.
2. Deserialize the error, i.e. every time we deal with RTK Query we have to call `deserialize` or `selectError` before calling `getErrorMessage` or any other functions.
3. In the `appBaseQuery` add the property `isAxiosError: true` to the serialized object if it is an `AxiosError`.

@BLoe @fregante thoughts, opinions?

Comment (fregante) [reply src/errors/networkErrorHelpers.ts:34]: Related:

- #3613
- https://github.com/reduxjs/redux-toolkit/issues/2392

According to RTK authors, I think we're dealing with errors the wrong way. If fixable, it'd be great to fix this sooner rather than later, so we don't have to patch all of these pieces of code.

Comment (twschiller) [reply src/errors/networkErrorHelpers.ts:34]: @BALEHOK my vote is to check for the name AxiosError. Could be low effort/a win to set `isAxiosError: true` in appBaseQuery if the error is an Axios error

I don't want to re-open the RTK error handling can of worms right now. We have bigger fish to fry (e.g., Page Editor rendering/performance). Happy to re-prioritize as necessary

Comment (BLoe) [reply src/errors/networkErrorHelpers.ts:34]: I agree with Federico that it'd be nice to fix this kind of issue sooner rather than later, but yeah, the name check is ok for now I suppose. Can we create a ticket to track that and link it in the code there? Or at least leave a comment in the code there about fixing RTK Query error serialization at some point?

Comment (twschiller) [reply src/errors/networkErrorHelpers.ts:34]: @fregante had created one here: https://github.com/pixiebrix/pixiebrix-extension/issues/4032 🙏 

We also have historic tickets/PRs from the existing efforts, I can help look those up as needed

Comment (fregante) [reply src/errors/networkErrorHelpers.ts:34]: I forgot I had opened https://github.com/pixiebrix/pixiebrix-extension/issues/4032 😅 

I added more related links to that issue, including this discussions

Comment (BALEHOK) [reply src/errors/networkErrorHelpers.ts:37]: I added a comment and a link to the issue.

**Review by BLoe — APPROVED** [review]
